raiden/transfer/architecture.py

```diff
@@ -264,6 +280,13 @@ def dispatch(self, state_change: StateChange) -> List[Event]:
 
         assert isinstance(iteration, TransitionResult)
 
+        # if any balance proofs have changed, create events
+        for balance_proof in self.detect_balance_proof_change(
+                self.current_state,
+                iteration.new_state,
+        ):
+            iteration.events.append(EventNewBalanceProofReceived(balance_proof))
```

Comment:
This is no place for this logic, the state manager does all it needs to do already. You can just call `detect_balance_proof_change` after a state change is processed.